plugins/catalyst/python/src/mqt/catalyst/__init__.py

```diff
@@ -0,0 +1,93 @@
+
+from __future__ import annotations
+
+from importlib.metadata import PackageNotFoundError, distribution
+from pathlib import Path
+
+import pennylane as qml
+
+from catalyst.passes import PassPlugin
+
+
+def get_catalyst_plugin_abs_path() -> Path:
+    """Locate the mqt-catalyst-plugin library.
+
+    Returns:
+        The absolute path to the mqt-catalyst-plugin library.
+
+    Raises:
+        FileNotFoundError: If the include directory is not found.
+        ImportError: If mqt-catalyst-plugin is not installed.
+    """
+    try:
+        dist = distribution("mqt-catalyst-plugin")
+        # Check for the plugin in the mqt-core package on Linux
+        catalyst_plugin_abs_path = Path(dist.locate_file("mqt/catalyst/lib/mqt-catalyst-plugin.so"))
+        if catalyst_plugin_abs_path.exists() and catalyst_plugin_abs_path.is_file():
+            return catalyst_plugin_abs_path
+        # Check for the plugin in the mqt-core package on macOS
+        catalyst_plugin_abs_path = Path(dist.locate_file("mqt/catalyst/lib/mqt-catalyst-plugin.dylib"))
+        if catalyst_plugin_abs_path.exists() and catalyst_plugin_abs_path.is_file():
+            return catalyst_plugin_abs_path
+        msg = "mqt-catalyst-plugin library not found."
+        raise FileNotFoundError(msg)
+    except PackageNotFoundError:
+        msg = "mqt-catalyst-plugin not installed, installation required to access the include files."
+        raise ImportError(msg) from None
+
+
+def name2pass(_name: str) -> tuple[Path, str]:
+    return get_catalyst_plugin_abs_path(), "mqt-core-round-trip"
+
+
+def MQTCoreRoundTrip(*flags, **valued_options):
+    def add_pass_to_pipeline(**kwargs):
+        pass_pipeline = kwargs.get("pass_pipeline", [])
+        pass_pipeline.append(
+            PassPlugin(
+                get_catalyst_plugin_abs_path(),
+                "mqt-core-round-trip",
+                *flags,
+                **valued_options,
+            )
+        )
+        return pass_pipeline
+
+    def decorator(qnode):
+        if not isinstance(qnode, qml.QNode):
+            # Technically, this apply pass is general enough that it can apply to
+            # classical functions too. However, since we lack the current infrastructure
+            # to denote a function, let's limit it to qnodes
+            msg = f"A QNode is expected, got the classical function {qnode}"
+            raise TypeError(msg)
+
+        def qnode_call(*args, **kwargs):
+            kwargs["pass_pipeline"] = add_pass_to_pipeline(**kwargs)
+            return qnode(*args, **kwargs)
+
+        return qnode_call
+
+    # When the decorator is used without ()
+    if len(flags) == 1 and isinstance(flags[0], qml.QNode):
+        qnode = flags[0]
+
+        def qnode_call(*args, **kwargs):
+            kwargs["pass_pipeline"] = add_pass_to_pipeline(**kwargs)
+            return qnode(*args, **kwargs)
+
+        return qnode_call
+
+    # When the decorator is used with ()
+    return decorator
```

Comment:
Shouldn't this now expose the individual conversion passes and all of the MQT passes?